### PR TITLE
first work on creating an AJAX call for importing list

### PR DESCRIPTION
**Pull request by eddiebroad**
got import list reload when you make it appear/reappear ;
the idea/goal is it should re-load data for importing when it disappears/reappears

### CODE REVIEW

**Comment by dmohs** [comment]
A portion of this PR is whitespace changes that are inconsistent with our style conventions. I think they can mostly be fixed by Settings -> Editor -> Code Style -> Clojure -> General -> Default to Only Indent. Please try that and reformat the code and then I'll take another look.

**Comment by eddiebroad** [comment]
@dmohs thanks for this note

**Comment by eddiebroad** [comment]
1) I took an action (re-re-base) that caused a conflict message to go away. 
2) I eliminated double-calling of AJAX via if-statement switching on (:import-overlay-shown? @state)

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs [hunk 1]
```diff
@@ -1,7 +1,8 @@
 (ns org.broadinstitute.firecloud-ui.page.workspace.method-config-importer
   (:require
     [dmohs.react :as react]
-    [org.broadinstitute.firecloud-ui.utils :as utils]
+    [clojure.string :refer [join]]
+    [org.broadinstitute.firecloud-ui.utils :as utils  ]
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs]: Remove whitespace after utils.

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs [hunk 2]
```diff
@@ -32,59 +33,85 @@
...
 (react/defc ImportWorkspaceMethodsConfigurationsList
-  {:render
-   (fn [{:keys [props]}]
+  {:component-did-mount
+   (fn [{:keys [state props]}]
+     (utils/ajax-orch
+       (str "/workspaces/" (:selected-workspace-namespace props) "/" (:selected-workspace props) "/methodconfigs")
+       {:on-done (fn [{:keys [success? xhr]}]
+                   (if success?
+                     ((fn []
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs]: This is just creating a function and calling it. If you want to execute multiple statements, use `do`:
```
(if success?
  (do
    (swap! state...)
```

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs [hunk 3]
```diff
@@ -32,59 +33,85 @@
...
+                         :cell-renderer (fn [row-num conf]
+                                          (cell (utils/map-to-string (conf "outputs"))))}
+                        {:header-component (header "Prerequisites") :starting-width 200
+                         :cell-renderer (fn [row-num conf] (join "," (conf "prerequisites")))}]
+              :data (:method-confs @state)
+              :row-props (fn [row-num conf]
+                           {:style {:fontSize "80%" :fontWeight 500
+                                    :paddingTop 10 :paddingBottom 7
+                                    :backgroundColor (if (even? row-num)
+                                                       (:background-gray style/colors) "#fff")}})})])
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs]: Add a newline before "#fff" to make it clearer that it is the "else" clause here.

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs [hunk 4]
```diff
@@ -105,13 +132,17 @@
 
 
 (defn- render-import-overlay [state]
-  (let [clear-import-overlay #(swap! state assoc :import-overlay-shown? false)]
+  (let [
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs]: No newline after [.

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs [hunk 5]
```diff
@@ -105,13 +132,17 @@
...
+        clear-import-overlay #(swap! state assoc :import-overlay-shown? false)
+        render-import-overlay-react (react/create-class { :render (fn
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs]: Re-indent. You could do:
```
(react/create-class
  {:render (fn ...
```

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs [hunk 6]
```diff
@@ -105,13 +132,17 @@
...
+            [{:keys [state props]}]
+               [ImportWorkspaceMethodsConfigurationsList])})]
     [:div {:style (modal-import-background state)
            :onKeyDown (common/create-key-handler [:esc] clear-import-overlay)}
      [:div {:style modal-import-content}
       [:div {:style {:position "absolute" :right 2 :top 2}}
        [:div {:style {:backgroundColor (:button-blue style/colors) :color "#fff"
-                      :padding "0.5em" :cursor "pointer"}
+                      :padding         "0.5em" :cursor "pointer"}
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs]: Revert this line.

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs [hunk 7]
```diff
@@ -119,9 +150,11 @@
                      :padding "20px 48px 18px"}}
        [:div {:style {:fontSize 24 :align "center" :textAlign "center" :paddingBottom "0.5em"}}
         "Select Method Configurations For Import"]
-       [ImportWorkspaceMethodsConfigurationsList {:method-confs (utils/parse-json-string
-                                                                  (utils/->json-string
-                                                                    (create-mock-methodconfs-import)))}]
+         [
```
Comment (dmohs) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs]: Don't think you need these brackets.

**Comment by dmohs** [comment]
Getting very close. Needs a few more cleanups.

File: src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs (hunk 7 above)
Comment (eddiebroad) [inline src/cljs/org/broadinstitute/firecloud_ui/page/workspace/method_config_importer.cljs]: Are you referring to the outer-most "[" and "]" here containing ONLY an if-statement and its true and false blocks and an "x" placeholder?
If yes, then it seems they are needed.  I tried with and without the brackets and without them the import-overlay doesn't use the render-import-overlay-react created in the "let" about 20 lines above.  Maybe you want a refactor with a larger scope (than the brackets)?

**Comment by dmohs** [comment]
:+1:

**Pull request merged**